### ISSUE TITLE
problem of extracting flow feature

### ISSUE THREAD

**Issue by hao-zheng-uoa** [LeYangNwpu/ExtractVideoFeature#2]
Thank you for providing such a good repo. It helps me a lot. 
I have successfully got the rgb features, but when I change the mode to flow, I met problems.
when I runned `python demo_extract_i3d_feature.py -mode flow -weight_file ./models/flow_imagenet.pt`
I got the error: `RuntimeError: Given groups=1, weight of size [64, 2, 7, 7, 7], expected input[8, 3, 21, 229, 229] to have 2 channels, but got 3 channels instead`
Can you help me out here?
Thank you again.

**Comment by bibofu** [comment]
I met this problem too....

**Comment by duckburgerhill** [comment]
> Thank you for providing such a good repo. It helps me a lot. I have successfully got the rgb features, but when I change the mode to flow, I met problems. when I runned `python demo_extract_i3d_feature.py -mode flow -weight_file ./models/flow_imagenet.pt` I got the error: `RuntimeError: Given groups=1, weight of size [64, 2, 7, 7, 7], expected input[8, 3, 21, 229, 229] to have 2 channels, but got 3 channels instead` Can you help me out here? Thank you again.

你看一看是不是你的checkpoint那个pt文件没有换成flow_imagenet.pt

**Comment by duckburgerhill** [comment]
> Thank you for providing such a good repo. It helps me a lot. I have successfully got the rgb features, but when I change the mode to flow, I met problems. when I runned `python demo_extract_i3d_feature.py -mode flow -weight_file ./models/flow_imagenet.pt` I got the error: `RuntimeError: Given groups=1, weight of size [64, 2, 7, 7, 7], expected input[8, 3, 21, 229, 229] to have 2 channels, but got 3 channels instead` Can you help me out here? Thank you again.

Did you solve it?

**Comment by hao-zheng-uoa** [comment]
> > Thank you for providing such a good repo. It helps me a lot. I have successfully got the rgb features, but when I change the mode to flow, I met problems. when I runned `python demo_extract_i3d_feature.py -mode flow -weight_file ./models/flow_imagenet.pt` I got the error: `RuntimeError: Given groups=1, weight of size [64, 2, 7, 7, 7], expected input[8, 3, 21, 229, 229] to have 2 channels, but got 3 channels instead` Can you help me out here? Thank you again.
> 
> Did you solve it?

No, I just quit using optical flow.

**Comment by duckburgerhill** [comment]
> 
我不知道我这样回答对不对。我觉得你应该使用flow帧，而不是rgb帧。因为它说channel应该是2，就是光流的x,y。你可以去用mmaction2中的工具去提取flow帧。它是x,y两个方向的jpg

**Comment by VividLe** [comment]
Yes. We should first calcualte optical flow from video frames, then extract flow features.
